src/components/SearchBar/SearchBarMobile.tsx

```diff
@@ -0,0 +1,253 @@
+import { Input } from '@components/ui/input';
+import { Popover, PopoverTrigger } from '@components/ui/popover';
+import { H2, Small } from '@components/ui/typography';
+import cn from '@lib/utils';
+import { Dialog, DialogTrigger } from '@radix-ui/react-dialog';
+import { motion } from 'framer-motion';
+import { HashIcon, LucideIcon, MapIcon, SearchIcon, XIcon } from 'lucide-react';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+import { useRef, useState } from 'react';
+import MenuItemContainer from './MenuItemContainer';
+import useDimensions from './use-dimensions';
+
+type SearchBarMobileProps = {
+  className: string;
+  activityPictures: Array<{
+    thumbnail: string;
+    url: string;
+    description: string;
+  }>;
+  activityKeywords: Array<{
+    url: string;
+    keyword: string;
+  }>;
+  locations: Array<{
+    url: string;
+    text: string;
+  }>;
+  categories: Array<{
+    icon: LucideIcon;
+    url: string;
+    text: string;
+  }>;
+  debugMode: boolean;
+};
+
+const locationMenu = {
+  open: (height = 1000) => ({
+    clipPath: `circle(${height * 2 + 200}px at 40px 800px)`,
+    transition: {
+      type: 'spring',
+      stiffness: 10,
+      restDelta: 2,
+    },
+  }),
+  closed: {
+    // TODO: use mouse event to get cursor position
+    clipPath: 'circle(0px at 40px 800px)',
+    transition: {
+      delay: 0.5,
+      type: 'spring',
+      stiffness: 400,
+      damping: 40,
+    },
+  },
+};
+
+const categoryMenu = {
+  open: (height = 1000) => ({
+    clipPath: `circle(${height * 2 + 200}px at 340px 800px)`,
+    transition: {
+      type: 'spring',
+      stiffness: 10,
+      restDelta: 2,
+    },
+  }),
+  closed: {
+    // TODO: use mouse event to get cursor position
+    clipPath: 'circle(0px at 340px 800px)',
+    transition: {
+      delay: 0.5,
+      type: 'spring',
+      stiffness: 400,
+      damping: 40,
+    },
+  },
+};
+
+const SearchBarMobile = ({
+  className,
+  activityPictures,
+  activityKeywords,
+  locations,
+  categories,
+  debugMode,
+}: SearchBarMobileProps) => {
+  const [isCategoryMenuOpen, setCategoryMenuOpen] = useState(false);
+  const [isLocationMenuOpen, setLocationMenuOpen] = useState(false);
+  const containerRef = useRef(null);
+  const { height } = useDimensions(containerRef);
+  const router = useRouter();
+
+  return (
+    <Dialog defaultOpen={debugMode}>
+      <DialogTrigger
+        className={cn(
+          'fixed bottom-0 right-0 z-10 flex flex-col items-center justify-center gap-4 bg-primary p-7 font-medium text-white',
+          className
+        )}
+      >
+        <SearchIcon className="size-6" />
+        <p>搜尋活動</p>
+      </DialogTrigger>
+      <DialogContent hideCloseButton className="block h-svh w-screen p-0">
+        <DialogHeader>
+          <DialogTitle asChild className="flex items-end justify-between">
+            <div>
+              <H2 className="ml-3 text-primary">搜尋活動</H2>
```

Comment:
```suggestion
              <H2 className="mb-1 ml-3 text-primary">搜尋活動</H2>
```

Reply:
fix one this commit: 865720f6b51e974e42f94b0d75be4cf0a21c76ba